### ISSUE TITLE
Type declared a "type: nil" information is lost on resolution

### ISSUE THREAD

**Issue by postatum** [aml-org/amf#555]
amf-client-js@3.4.1

When type is defined as `type: nil` or an explicit `nil`, it's information is lost when model is resolved and it's impossible to find it by searching through shapes by name or id.

E.g. running code [1] produces [2]. Note how `Resolved:` output lacks `Dog http://a.ml/amf/default_document#/declarations/types/nil/Dog` while `Cat http://a.ml/amf/default_document#/declarations/types/scalar/Cat` is still there.

[1]
```js
const amf = require('amf-client-js')

const ramlStr = `
  #%RAML 1.0

  types:
    Cat: string
    Dog: nil

  /cats:
    get:
      responses:
        200:
          body:
            application/json:
              type: Cat
        400:
          body:
            application/json:
              type: Dog
`

async function main () {
  await amf.AMF.init()
  const parser = amf.AMF.raml10Parser()
  let shapes

  const model = await parser.parseStringAsync(ramlStr)
  shapes = model.findByType('http://a.ml/vocabularies/shapes#Shape')
  console.log('\nNot resolved:')
  shapes.forEach(shape => {
    console.log(shape.name.value(), shape.id)
  })

  const resolved = amf.AMF.resolveRaml10(model)
  shapes = model.findByType('http://a.ml/vocabularies/shapes#Shape')
  console.log('\nResolved:')
  shapes.forEach(shape => {
    console.log(shape.name.value(), shape.id)
  })
}

main()
```
[2]
```
$ node raml10.js 

Not resolved:
schema http://a.ml/amf/default_document#/web-api/end-points/%2Fcats/get/200/application%2Fjson/scalar/schema
Cat http://a.ml/amf/default_document#/declarations/types/scalar/Cat/linked_1
Cat http://a.ml/amf/default_document#/declarations/types/scalar/Cat
schema http://a.ml/amf/default_document#/web-api/end-points/%2Fcats/get/400/application%2Fjson/nil/schema
Dog http://a.ml/amf/default_document#/declarations/types/nil/Dog

Resolved:
Cat http://a.ml/amf/default_document#/declarations/types/scalar/Cat
schema http://a.ml/amf/default_document#/web-api/end-points/%2Fcats/get/400/application%2Fjson/nil/schema
```

**Comment by looseale** [comment]
Related JIRA issue: https://www.mulesoft.org/jira/browse/APIMF-1646

**Comment by looseale** [comment]
Fixed. To be released in the next version (9th October).
Already available in the snapshot version (3.6.0-SNAPSHOT.1).
